src/nvim/eval/funcs.c

```diff
@@ -2888,6 +2888,33 @@ static void f_getregtype(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
   rettv->vval.v_string = xstrdup(buf);
 }
 
+/// "getscriptinfo()" function
+static void f_getscriptinfo(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
+{
+  int i;
+  list_T *l;
+
+  tv_list_alloc_ret(rettv, script_items.ga_len);
+
+  l = rettv->vval.v_list;
+
+  for (i = 1; i <= script_items.ga_len; ++i) {
+    scriptitem_T *si = SCRIPT_ITEM(i);
+    dict_T *d;
+
+    if (si->sn_name == NULL) {
+      continue;
+    }
+
+    d = tv_dict_alloc();
+    tv_list_append_dict(l, d);
+    tv_dict_add_str(d, S_LEN("name"), si->sn_name);
+    tv_dict_add_nr(d, S_LEN("sid"), i);
+    // Vim9 autoload script (:h vim9-autoload), not applicable to Neovim.
+    tv_dict_add_bool(d, S_LEN("autoload"), false);
```

Comment:
For the record: our `scriptitem_T` does not even have the fields that Vim is using to calculate this. We would need to port at least patches 8.2.2222 and 8.2.4050, but this is for vim9script only.